processor/src/main/java/org/mapstruct/ap/model/source/SourceMethod.java

```diff
@@ -61,10 +64,14 @@
     private final Accessibility accessibility;
     private final List<Type> exceptionTypes;
     private final MapperConfig config;
+    private final MappingOptions mappingOptions;
+    private final List<SourceMethod> prototypeMethods;
```

Comment:
```prototypeMethods``` are all the available prototype methods in the current scope. If needed, the list ```applicablePrototypeMethods``` is initialized and contains only those that are applicable.